commcare_connect/opportunity/tasks.py

```diff
@@ -332,12 +332,18 @@ def bulk_approve_completed_work():
         for completed_work in completed_works:
             if completed_work.completed_count > 0:
                 approved_count = completed_work.approved_count
-                visits = completed_work.uservisit_set.values_list("status", "reason")
-                if any(status == "rejected" for status, _ in visits):
+                visits = completed_work.uservisit_set.values_list("status", "reason", "review_status")
+                if any(status == "rejected" for status, *_ in visits):
                     completed_work.update_status(CompletedWorkStatus.rejected)
-                    completed_work.reason = "\n".join(reason for _, reason in visits if reason)
-                elif all(status == "approved" for status, _ in visits):
+                    completed_work.reason = "\n".join(reason for _, reason, _ in visits if reason)
+                elif all(status == "approved" for status, *_ in visits):
                     completed_work.update_status(CompletedWorkStatus.approved)
+                if (
+                    access.opportunity.managed
+                    and not all(review_status == "agree" for *_, review_status in visits)
```

Comment:
nit: this should reference a constant like we do for the other status

Reply:
I will make this change once I can merge the main into my current PR. Some changes around the status updates from the main branch are not present on this PR causing issues whenever I change something.